File: shuffle-string/shuffle-string.js
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function(s, indices) {
    let tempObject = {};
    let arr_str = [...s];
    for (let i in indices) {
        tempObject[indices[i]] = arr_str[i];
    }
    let finalString = Object.values(tempObject).join('')
    return finalString;
};

File: shuffle-string/shuffle-string.py
class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        dict = {}
        
        for i in range(len(s)):
            key = indices[i]
            value = s[i]
            dict[key] = value
            
        
        sorted_dict = OrderedDict(sorted(dict.items()))
        

        final_string = ""
        for elem in sorted_dict.values():
            final_string += elem
        
        return final_string